pkg/addons/client.go

```diff
@@ -109,11 +116,22 @@ func (c *client) waitForHealthz() error {
 
 	for {
 		resp, err := cli.Do(req)
-		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
-			time.Sleep(time.Second)
-			continue
+		if err, ok := err.(*url.Error); ok {
```

Comment:
Create a helper to keep this loop cleaner
```go
func needsRetry(err error) bool {
	if err, ok := err.(*url.Error); ok {
		if err, ok := err.Err.(*net.OpError); ok {
			if err, ok := err.Err.(*os.SyscallError); ok {
				if err.Err == syscall.ENETUNREACH {
					return true
				}
			}
		}
		if err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF {
			return true
		}
		return false
	}
}
```

Reply:
yes, it was about time to do that ;-)